mars/session.py

```diff
@@ -190,6 +190,46 @@ def default_or_local(cls):
         cls._default_session = Session()
         return cls._default_session
 
+    def create_mutable_tensor(self, name, shape, dtype, *args, **kwargs):
+        from .tensor.core import MutableTensor, MutableTensorData
+        from .tensor.expressions.utils import create_fetch_tensor
+        self._ensure_local_cluster()
+        shape, dtype, chunk_size, chunk_keys = \
+                self._sess.create_mutable_tensor(name, shape, dtype, *args, **kwargs)
+        # Construct MutableTensor on the fly.
+        tensor = create_fetch_tensor(chunk_size, shape, dtype, chunk_keys)
+        return MutableTensor(chunk_size=chunk_size,
+                             data=MutableTensorData(_name=name, _op=None, _shape=shape, _dtype=dtype,
+                                                    _nsplits=tensor.nsplits, _chunks=tensor.chunks))
+
+    def get_mutable_tensor(self, name):
+        from .tensor.core import MutableTensor, MutableTensorData
+        from .tensor.expressions.utils import create_fetch_tensor
+        self._ensure_local_cluster()
+        shape, dtype, chunk_size, chunk_keys = self._sess.get_mutable_tensor(name)
+        # Construct MutableTensor on the fly.
+        tensor = create_fetch_tensor(chunk_size, shape, dtype, chunk_keys)
+        return MutableTensor(chunk_size=chunk_size,
+                             data=MutableTensorData(_name=name, _op=None, _shape=shape, _dtype=dtype,
+                                                    _nsplits=tensor.nsplits, _chunks=tensor.chunks))
+    def write_mutable_tensor(self, tensor, index, value):
+        self._ensure_local_cluster()
+        chunk_records_to_send = tensor._do_write(index, value)
+        return self._sess.send_chunk_records(tensor.name, chunk_records_to_send)
+
+    def seal(self, tensor):
+        from .tensor.expressions.utils import create_fetch_tensor
+        self._ensure_local_cluster()
+        chunk_records_to_send = tensor._do_flush()
+        self._sess.send_chunk_records(tensor.name, chunk_records_to_send)
+        shape, dtype, chunk_size, chunk_keys = self._sess.seal(tensor.name)
+        # Construct Tensor on the fly.
+        return create_fetch_tensor(chunk_size, shape, dtype, chunk_keys)
+
+    def _ensure_local_cluster(self):
+        from .deploy.local.session import LocalClusterSession
+        if not isinstance(self._sess, LocalClusterSession):
```

Comment:
So for now only local cluster is supported?

Reply:
The limitation is because we need the client can request the worker directly when writing to mutable tensor.

For `LocalSession`, I think there are no actors like `PlasmaStoreActor`? For `WebSession`, I haven't looked into the details, but I think many things would be different, compared with `LocalClusterSession`?

Reply:
I think we can focus on local cluster for this PR, and we need to modify the name to mark the limitation.

For local session, there is no plasma, because it's threaded. I think it's not so reasonable for local session, because user can use numpy instead, but if we want to make mutable tensor available for all schedulers, we could implement a naive one for local session.

As for web session, we need to add some RESTful interface.

I think we can add support for local and distributed schedulers in the next PR.